addons/web/static/src/views/fields/parsers.js

```diff
@@ -133,6 +133,9 @@ export function parseInteger(value) {
             throw new InvalidNumberError(`"${value}" is not a correct number`);
         }
     }
+    if (parsed % 1 || parsed < -2147483648 || parsed > 2147483647) {
```

Comment:
All three checks could be combined into `if (parsed | 0 !== parsed)` thanks to the fact that bitwise operations return 32 bits integers in javascript. Not sure if that would be considered too magical though

Reply:
> All three checks could be combined into `if (parsed | 0 !== parsed)` thanks to the fact that bitwise operations return 32 bits integers in javascript. Not sure if that would be considered too magical though

Hello ,
Already used in 15.0 so I used it [15.0](https://github.com/odoo/odoo/blob/15.0/addons/web/static/src/legacy/js/fields/field_utils.js#L700). I checked your suggestion but doesn't' working.

Reply:
You are right, it won't work because `!==` has a higher priority than `|`.
This should work however: `if ((parsed | 0) !== parsed)`.

But as you said, it might be better to keep it as it was in 15.0 then.